app/uk/gov/hmrc/gform/gform/DeclarationController.scala

```diff
@@ -246,11 +246,23 @@ class DeclarationController(
     val form: Form = cache.form.copy(
       formData = variadicFormData.toFormData
     )
+
+    val declarationSectionFieldIds: List[FormComponentId] = cache.formTemplate.destinations match {
+      case DestinationList(_, _, ds) => ds.fields.map(_.id)
```

Comment:
This should support revealing choice as well. You can use `childrenFormComponents` method on `FormComponent`